util/libcrypto.num

```diff
@@ -4379,3 +4379,7 @@ BN_priv_rand_range                      4322	1_1_1	EXIST::FUNCTION:
 ASN1_TIME_normalize                     4323	1_1_1	EXIST::FUNCTION:
 ASN1_TIME_cmp_time_t                    4324	1_1_1	EXIST::FUNCTION:
 ASN1_TIME_compare                       4325	1_1_1	EXIST::FUNCTION:
+ossl_ctype_check                        4326	1_1_1	EXIST::FUNCTION:
+ossl_tolower                            4327	1_1_1	EXIST::FUNCTION:
+ossl_toupper                            4328	1_1_1	EXIST::FUNCTION:
+ossl_isalnum                            4329	1_1_1	EXIST::FUNCTION:
```

Comment:
I've fixed the one instance in libssl that used this.
I've removed the test case since it fails to link - I'll work on putting it back in due course.
I've reverted the apps that use the isXXX macros.
I've removed the exports.

We can re-expose `ossl_ctype_check` if we think it is desirable (this does something similar to the `ossl_ctype` suggestion).  I'm happy to rename the former to the latter if it might help.